modules/gdscript/gdscript_formatter.cpp

```diff
@@ -0,0 +1,46 @@
+/**************************************************************************/
+/*  gdscript_formatter.cpp                                                */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifdef TESTS_ENABLED
+
+#include "gdscript_formatter.h"
+
+Error GDScriptFormatter::format(const String &p_code, String &r_formatted_code) {
+	if (p_code == "")
+		return OK;
+	if (p_code == "pass" || p_code == "pass\n") {
+		r_formatted_code = "pass\n";
+		return OK;
+	}
+	ERR_PRINT("Formatting failed.");
+	return FAILED;
```

Comment:
If we really want a no-op, then I suggest this.

```suggestion
	r_formatted_code = p_code;
	return OK;
```

The formatting shouldn't fail for code that wasn't formatted.

Reply:
Okay, let's start with an actual noop - it will be cleaner this way - good point.